### PR TITLE
refactor(components): migrate Card to Typescript

### PR DESCRIPTION
**Pull request by AlineLee**
Addresses: https://github.com/sumup-oss/circuit-ui/issues/516

## Approach and changes

- migrate Card component to TypeScript

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Comment by connor-baer** [comment]
@AlineLee Is this ready for review? :)

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -67,34 +79,8 @@ const spacingStyles = ({ theme, spacing }) => {
...
-
-/**
- * @component
- */
-export default Card;
+export const Card: FC<CardProps> = styled('div')<CardProps>(
+  baseStyles,
+  shadowStyles,
+  spacingStyles,
+);
```
Comment (connor-baer) [inline src/components/Card/Card.tsx:86]: Is there a reason why you removed the `shouldForwardProp` option? It still seems necessary to prevent invalid HTML attributes.

File: src/components/Card/components/Header/Header.tsx [hunk 2]
```diff
@@ -13,86 +13,85 @@
...
+  onClose,
   labelCloseButton,
-  tracking = {},
   ...props
 }) => (
-  <CardHeaderContainer {...props}>
+  <CardHeaderContainer
+    labelCloseButton={labelCloseButton}
+    tracking={tracking}
+    onClose={onClose}
```
Comment (connor-baer) [inline src/components/Card/components/Header/Header.tsx:81]: Why did you add these props to this element?

**Comment by AlineLee** [comment]
> @AlineLee Is this ready for review? :)

Yes!

File: src/components/Card/components/Header/Header.tsx [hunk 3]
```diff
@@ -13,86 +13,85 @@
...
+    {...props}
+  >
     {children}
-    {onClose && (
+    {onClose && labelCloseButton && (
```
Comment (connor-baer) [inline src/components/Card/components/Header/Header.tsx:85]: Love that this forces developers to provide the label. Great for accessibility 🙌

File: src/components/Card/components/Header/Header.tsx [hunk 4]
```diff
@@ -13,86 +13,85 @@
...
   justify-content: space-between;
   margin-bottom: ${theme.spacings.giga};
 `;
 
-const noHeadingStyles = ({ children }) =>
-  !children[0] &&
-  css`
-    label: card__header--no-heading;
-    justify-content: flex-end;
-  `;
```
Comment (connor-baer) [inline src/components/Card/components/Header/Header.tsx:37]: I assume you removed this because the `children` prop is required now? 

We still cannot assume that `children` will be defined since this isn't enforceable in JavaScript projects. 

Furthermore, should the `children` prop always be required? I can imagine a use case where the close button is needed, but no title for the card. What do you think?

File: src/components/NotificationList/NotificationList.tsx [hunk 5]
```diff
@@ -52,8 +52,9 @@ const baseStyles = ({ theme }: StyleProps) => css`
 
 const NotificationListWrapper = styled('ul')<{}>(baseStyles);
 
-// FIXME: Remove typecast once the Card has been migrated to TypeScript.
-const NotificationListCard = Card.withComponent('li') as any;
+// FIXME: how to use li: Card.withComponent('li');
+// const NotificationListCard = Card.withComponent('li');
+const NotificationListCard = styled(Card)<NotificationListProps>();
```
Comment (connor-baer) [inline src/components/NotificationList/NotificationList.tsx:57]: It looks like this comment has been resolved, hasn't it?

File: src/components/Card/components/Header/Header.tsx (hunk 2 above)
Comment (AlineLee) [reply src/components/Card/components/Header/Header.tsx:81]: 🤔 
I was trying to pass the attributes ahead .. so I included them to the `CardHeaderContainer`, to the `CardHeaderContainer` pass the attribute instead of just use directly int the `CardHeaderCloseButton`

File: src/components/Card/components/Header/Header.tsx (hunk 4 above)
Comment (AlineLee) [reply src/components/Card/components/Header/Header.tsx:37]: I fix the children, now It's not required 😄  and the style for this is back!

File: src/components/Card/Card.tsx (hunk 1 above)
Comment (AlineLee) [reply src/components/Card/Card.tsx:86]: Well I thought that using typescript the prop was not necessary anymore... but thinking better, one thing doesn't relate with the other 🤣

File: src/components/Card/components/Header/Header.tsx (hunk 3 above)
Comment (connor-baer) [reply src/components/Card/components/Header/Header.tsx:90]: I just realized that this is a breaking change 😞 The close button might disappear e.g. on the dashboard if the label is missing. We'll need to revert this until the next major version.

**Review by connor-baer — APPROVED** [review]
